### ISSUE TITLE
Last class reused because of common sub styles

### ISSUE THREAD

**Issue by ben-rogerson** [cristianbote/goober#199]
While testing the recent changes to Goober I came across a strange bug.

The class of an element gets reused for another because they share a group of styles.
The common styles are shown here, named `stylesBase`:

```js
const stylesBase = { padding: '20px', color: 'white' }

// This is green, looks like the class is being reused
// for both the 'Button' and the 'div' components
const Button = styled('button')(stylesBase)

const App = ({ isPrimary = true }) => (
  <div>
    <Button>This shouldn't be green</Button>

    <div
      className={css([
        stylesBase,
        isPrimary && { 'background-color': 'green' }
      ])}
    >
      a div
    </div>
  </div>
)
```

Take a look at [the demo on codesandbox](https://codesandbox.io/s/goober-issue-example-jyed1?file=/src/index.js)

**Comment by cristianbote** [comment]
Yikes! looking into it. Thank you!

**Comment by cristianbote** [comment]
Ah, indeed. So the `Object.assign` from that reduce takes the first object and keeps the reference and does an mutation on it with the `backgroundColor`. Will open up a PR to fix it.

**Comment by cristianbote** [comment]
Published in `2.0.15`. Thank you!
